compiler/rustc_mir_transform/src/remove_zsts.rs

```diff
@@ -63,3 +51,80 @@ fn maybe_zst(ty: Ty<'_>) -> bool {
         _ => false,
     }
 }
+
+impl<'tcx> Replacer<'_, 'tcx> {
+    fn is_zst(&self, ty: Ty<'tcx>) -> bool {
```

Comment:
nit: call it `known_to_be_zst`, as it could be a parameter or sth that we can't compute a layout for